website/docs/metaprogramming-plugins.md

```diff
@@ -0,0 +1,124 @@
+---
+id: metaprogramming-plugins
+title: Metaprogramming plugins
```

Comment:
What do you think about

```yaml
id: dsls
title: Plugins for DSLs and Metaprogramming
sidebar_label: DSL Plugins
```

?

Reply:
I went with `metaprogramming` since I don't consider things like `attr_reader` and `delegate` to be DSLs. Also, one of the target audiences for this page is someone who's shopping for a type checker and wanted some questions answered. In my head, said person would search for `metaprogramming` rather than DSL.